src/pages/home/report/ReportActionCompose.js

```diff
@@ -205,6 +210,9 @@ class ReportActionCompose extends React.Component {
     }
 
     setIsFullComposerAvailable(isFullComposerAvailable) {
+        if (this.state.isFullComposerAvailable === isFullComposerAvailable) {
```

Comment:
Does this check do anything? I always thought that React was smart enough to do this check internally and not trigger the re-render if the state doesn't actually change.

Reply:
A quick Google search seems to reaffirm my understanding of this:

> For new React developers, it is common to think that setState will trigger a re-render. But this is not always the case. If the value doesn’t change, React will not trigger a re-render.

Reply:
Not sure where that quote is from, but its wrong 😅 It might be true if it's a `PureComponent`, however it isn't. So, another approach would be to turn it into a `PureComponent` if we feel that's cleaner?

Reply:
Yes, let's use `PureComponent` or `shouldComponentUpdate` instead. That feels cleaner to me.

Reply:
changed to pure component!